app/src/components/submissionPage/SnapList.tsx

```diff
@@ -64,12 +59,40 @@ const SnapList: React.FunctionComponent<SnapListProps> = ({ snaps }) => {
         });
     }
 
+    function getNameFromTag(tag: string): string {
+        const namePart = tag.match(fullNameRegex)[0];
+        return namePart.substring(1, namePart.length - 1);
+    }
+
+    const formattedSnapBody = (body: string) => {
```

Comment:
This whole method has the potential get slightly gnarly. I would like to see it in its own module, with several unit tests to help us catch the edge cases.